### ISSUE TITLE
emoji representation in search bar

### ISSUE THREAD

**Issue by Ingegneus** [federico-terzi/espanso#1972]
### Description

I am working on an emoji picker package, but the emoji representation in the search bar only allows the black/white emojis. could this be changed somehow? also it would be neat to change the emoji font, so that all the newer emojis can be represented as well

### Motivation

I like the windows emoji picker, but it is limited to the windows emojis. i wanted to create an alternative

### Have you tried building it? Or would you like it?

- [ ] I'd be willing to contribute this feature to Espanso myself.

**Comment by smeech** [comment]
There are several emoji packages in espanso/hub - do they have the same behaviour?

**Comment by Ingegneus** [comment]
yes they do. I guess it has something to do with how espanso renders the font

**Comment by joh6nn** [comment]
Possibly a missing emoji font, or Espanso is having trouble finding it for some reason🤔

A quick google suggests that this is the default emoji font on Windows: https://learn.microsoft.com/en-us/typography/font-list/segoe-ui-emoji . Can you check and see if that's installed?

**Comment by smeech** [comment]
I agree - I suspect it's down to the installed fonts. Here (Linux Mint Xfce) I get coloured emojis in `actually-all-emojis` in the search bar, except for a few like ☺ and ☹, which also appear monochrome in my character map.

**Comment by ad08fee3** [comment]
_(See issue that seems related: #1972)_

**Comment by ad08fee3** [comment]
I looked into this a bit today, and I think I know the root cause. I'm just not sure how easily it can be fixed.

This forum discussion explains it better than I can: https://forums.wxwidgets.org/viewtopic.php?t=51209

**TLDR:** Espanso uses wxWidgets to render the GUI. The Windows operating system has two different ways to rendering text, and the one that wxWidgets uses is called "GDI" and does _not_ support color emojis.

To allow for color emoji support, there may be a way to use Windows' other method of rendering fonts by changing how we use wxWidgets to render the search results GUI. This is not an issue unique to Espanso ([Issue 1](https://github.com/wxWidgets/wxWidgets/issues/19275), [forum post 2](https://groups.google.com/g/wx-users/c/Ktfzcj3RAwI?pli=1), etc).

Edit: The [root cause in wxWidgets](https://github.com/wxWidgets/wxWidgets/issues/19275) doesn't seem too hard, if somebody wants to take a crack at it.

**Comment by ad08fee3** [comment]
Woohoo! Theoretically **[wxWidgets just merged support](https://github.com/wxWidgets/wxWidgets/issues/19275)** for color emojis on Windows!

It looks like it's [slated for release on **2026-07-07**](https://github.com/schiele/wxWidgets/commit/670835aec7e54b9e9d5a50d6a1aea1e8ba0bcdb2#diff-b2ef97476784dbcada96f8b5529433a95ef01c3b8f3ec4b771c8b109b6becf87L258).

So to close out this issue all we must do is upgrade Espanso to version 3.3.3 once it releases. 🎉

**Comment by rorotheworld** [comment]
It did in fact [release](https://github.com/wxWidgets/wxWidgets/releases/tag/v3.3.3).

**Comment by ad08fee3** [comment]
@smeech is there a defined process for doing dependency updates? Is there a lot of validation work to do?
